cmd/nitro/nitro.go

```diff
@@ -267,6 +270,16 @@ func mainImpl() int {
 			flag.Usage()
 			log.Crit("error generating HMAC key for Espresso Caff node", "err", err)
 		}
+		privHex := hex.EncodeToString(key.D.Bytes())
+		// This will be used by the hyperlane validator
+		os.Setenv("SNAPSHOT_PRIVATE_KEY", privHex)
+		if nodeConfig.ParentChain.ID != 0 {
```

Comment:
I dont understand why we need this check to see if parent chain id is zero or not